libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx

```diff
@@ -116,148 +103,130 @@ export const handleJobSubmit = (data: FieldValues, application: ApplicationEntit
   }
 }
 
+export const handleHelmSubmit = (data: FieldValues, helm: Helm) => {
+  const sourceProvider: 'HELM_REPOSITORY' | 'GIT' = data['source_provider']
+  const source = match(sourceProvider)
+    .with('GIT', () => {
+      const gitToken = getGitTokenValue(data['provider'] ?? '')
+
+      return {
+        git: {
+          git_repository: {
+            url: buildGitRepoUrl(gitToken?.type ?? data['provider'] ?? '', data['repository']),
+            branch: data['branch'],
+            root_path: data['root_path'],
+            git_token_id: gitToken?.id,
+          },
+        },
+      }
+    })
+    .with('HELM_REPOSITORY', () => ({
+      repository: {
+        repository: {
+          id: data['repository'],
+        },
+        chart_name: data['chart_name'],
+        chart_version: data['chart_version'],
+      },
+    }))
+    .exhaustive()
+
+  return {
+    ...helm,
+    name: data['name'],
+    description: data['description'],
+    source,
+    allow_cluster_wide_resources: data['allow_cluster_wide_resources'],
+    arguments: JSON.parse(data['arguments']),
+    timeout_sec: parseInt(data['timeout_sec'], 10),
+    auto_deploy: data['auto_deploy'] ?? false,
+  }
+}
+
 export function PageSettingsGeneralFeature() {
-  const { organizationId = '', projectId = '', environmentId = '', applicationId = '' } = useParams()
-  const queryClient = useQueryClient()
-  const dispatch = useDispatch<AppDispatch>()
-  const navigate = useNavigate()
-  const application = useSelector<RootState, ApplicationEntity | undefined>(
-    (state) => getApplicationsState(state).entities[applicationId],
-    (a, b) =>
-      a?.name === b?.name &&
-      a?.description === b?.description &&
-      a?.build_mode === b?.build_mode &&
-      a?.buildpack_language === b?.buildpack_language &&
-      a?.dockerfile_path === b?.dockerfile_path
-  )
+  const { organizationId = '', environmentId = '', applicationId = '' } = useParams()
 
   const { data: organization } = useOrganization({ organizationId })
-  const loadingStatus = useSelector((state: RootState) => getApplicationsState(state).loadingStatus)
+  const { data: service } = useService({ environmentId, serviceId: applicationId })
+  const { mutate: editService, isLoading: isLoadingEditService } = useEditService({ environmentId })
+
+  const helmRepository = match(service)
+    .with({ serviceType: 'HELM', source: P.when(isHelmRepositorySource) }, ({ source }) => source.repository)
+    .otherwise(() => undefined)
+  const helmGit = match(service)
+    .with({ serviceType: 'HELM', source: P.when(isHelmGitSource) }, ({ source }) => source.git?.git_repository)
+    .otherwise(() => undefined)
 
   const methods = useForm({
     mode: 'onChange',
+    defaultValues: {
```

Comment:
You can build `defaultValues` using `match` to avoid `as` cast everywhere and be sure about your type